src/main/java/org/skriptlang/skript/elements/expressions/ExprRotateQuaternion.java

```diff
@@ -23,35 +23,39 @@
 import org.eclipse.jdt.annotation.Nullable;
 import org.joml.Quaternionf;
 import org.joml.Vector3f;
+import org.skriptlang.skript.lang.util.JomlBukkitUtils;
 
 import ch.njol.skript.Skript;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.lang.util.SimpleExpression;
 import ch.njol.util.Kleenean;
 
-@Name("Rotate Quaternion")
+@Name("Rotate Quaternion/Vector")
 @Description({
-	"Rotates a Quaternion around an axis or a vector and by a set amount of degrees.",
-	"When you use a vector, the values will be used to rotate the x, y and z."
+	"Rotates a Quaternion around an axis a set amount of degrees or rotated by a vector.",
+	"When you use a vector, the values will be used to rotate the x, y and z.",
+	"Vector numbers will get converted from a double down to a float."
 })
-@Examples("rotate {_quaternion} around vector from 1, 0, 0 at player by 90 degrees")
+@Examples("set {_new} to {_quaternion} rotated by vector(1, 0, 0)")
+@RequiredPlugins("Spigot 1.19.4+")
 @Since("INSERT VERSION")
-public class ExprRotateQuaternion extends SimpleExpression<Quaternionf> {
+public class ExprRotateQuaternion extends SimpleExpression<Object> {
 
 	static {
 		if (Skript.isRunningMinecraft(1, 19, 4))
-			Skript.registerExpression(ExprRotateQuaternion.class, Quaternionf.class, ExpressionType.SIMPLE,
-					"%quaternions% rotated (around|on) [the] (:x|:y|:z)-axis by %number% [degrees|radians]",
+			Skript.registerExpression(ExprRotateQuaternion.class, Object.class, ExpressionType.SIMPLE,
+					"%quaternions/vectors% rotated (around|on) [the] (:x|:y|:z)-axis by %number% [degrees|radians]",
 					"%quaternions% rotated by %vector%");
```

Comment:
I don't really like this option at all. It's strange to use a vector as a rotation. Three number inputs would be better imo.

Reply:
That would destory parsing time.

Reply:
How so? It's used in both location and vector creation expressions without much trouble. You could add words to clarify too:
```%quaternions% rotated by %number% [degrees] [on x axis]...``` or so on, though that seems a bit clunky.